api/src/pcapi/core/educational/api.py

```diff
@@ -364,3 +372,56 @@ def notify_educational_redactor_on_educational_offer_or_stock_edit(
                 "data": data.dict(),
             },
         )
+
+
+def create_collective_stock(
+    stock_data: EducationalStockCreationBodyModel, user: User, *, legacy_id: Optional[int] = None
+) -> Optional[CollectiveStock]:
+    from pcapi.core.offers.api import _update_offer_fraud_information
+
+    offer_id = stock_data.offer_id
+    beginning = stock_data.beginning_datetime
+    booking_limit_datetime = stock_data.booking_limit_datetime
+    total_price = stock_data.total_price
+    number_of_tickets = stock_data.number_of_tickets
+    educational_price_detail = stock_data.educational_price_detail
+
+    if legacy_id:  # FIXME remove legacy support layer
+        collective_offer = (
+            CollectiveOffer.query.filter_by(offerId=offer_id)
+            .options(joinedload(CollectiveOffer.collectiveStock))
+            .one_or_none()
+        )
+        if not collective_offer:
+            return None
+    else:
+        collective_offer = (
+            CollectiveOffer.query.filter_by(id=offer_id).options(joinedload(CollectiveOffer.collectiveStock)).one()
+        )
+    if collective_offer.collectiveStock:
+        raise exceptions.CollectiveStockAlreadyExists()
+
+    offer_validation.check_validation_status(collective_offer)
+    if booking_limit_datetime is None:
+        booking_limit_datetime = beginning
+
+    collective_stock = CollectiveStock(
+        collectiveOffer=collective_offer,
+        beginningDatetime=beginning,
+        bookingLimitDatetime=booking_limit_datetime,
+        price=total_price,
+        numberOfTickets=number_of_tickets,
+        priceDetail=educational_price_detail,
+        stockId=legacy_id if legacy_id else None,  # FIXME remove legacy support layer
+    )
+    repository.save(collective_stock)
+    logger.info("Collective stock has been created", extra={"collective_offer": collective_offer.id})
```

Comment:
Je sais pas pourquoi on décide de logger l'id de l'offre, on peut la récupérer via le stock.
Au mieux on loggue les deux mais le choix de logger uniquement l'id de l'offre... 🤔 

```suggestion
    logger.info("Collective stock has been created", extra={"collective_offer": collective_offer.id, "collective_stock_id": collective_stock.id})
```